pkg/controller/stacks/install/stackinstall.go

```diff
@@ -167,22 +168,52 @@ func (f *handlerFactory) newHandler(ctx context.Context, ext v1alpha1.StackInsta
 // ************************************************************************************************
 func (h *stackInstallHandler) sync(ctx context.Context) (reconcile.Result, error) {
 	if h.ext.StackRecord() == nil {
+		// If we observe the Stack, InstallJob succeeded and we're done.
+		nn := types.NamespacedName{Name: h.ext.GetName(), Namespace: h.ext.GetNamespace()}
+		s := &v1alpha1.Stack{}
+
+		if err := h.kube.Get(ctx, nn, s); err != nil && !kerrors.IsNotFound(err) {
+			return fail(ctx, h.kube, h.ext, err)
+		} else if err != nil {
+			// Set a reference to the Stack record in StackInstall status
+			// Some stacksinstalls may not produce Stacks, rather StackDefinitions
+			// whose controllers will ultimately produce
+			h.ext.SetStackRecord(&corev1.ObjectReference{
+				APIVersion: s.APIVersion,
+				Kind:       s.Kind,
+				Name:       s.Name,
+				Namespace:  s.Namespace,
+				UID:        s.ObjectMeta.UID,
+			})
+			h.ext.SetConditions(runtimev1alpha1.Available(), runtimev1alpha1.ReconcileSuccess())
+
+			return requeueOnSuccess, h.kube.Status().Update(ctx, h.ext)
```

Comment:
Unless there is a fatal configuration, we always want the Stacks and StackInstalls controllers to be requeued so that our `Reconcile` and `sync()` functions get another opportunity to act on the resource since in both cases they are dependent on outside resources that are not event or watch driven.

https://github.com/crossplaneio/crossplane/blob/18ebbb0b800e6b70886809997dc4378cc9379ded/pkg/controller/stacks/stack/stack.go#L75-L76

_`requeueAfterOnSuccess` is a duration of 10s_

Reconcile Result is defined upstream: https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/reconcile/reconcile.go#L25-L33

From a simpler review premise, this is what were already doing here, https://github.com/crossplaneio/crossplane/pull/1217/files#diff-a2cd537f6531321fbef9a92d96b2c425L272 where the StackInstall asserted that a completed InstallJob was a complete StackInstall.  Now that the StackInstall has to update the StackRef itself, this call moved around.

I suppose https://github.com/crossplaneio/crossplane/pull/1217/files#diff-a2cd537f6531321fbef9a92d96b2c425R306 could be requeued without a timer (`resultRequeue`), since a complete installjob is no longer a complete stackinstall.  This could save up to 10s.

I would want to confirm this and see other examples of Crossplane returning reconcile results like this in a partial success condition before making that change.